crates/checkpoint-downloader/src/downloader.rs

```diff
@@ -231,7 +234,7 @@ impl ParallelCheckpointDownloaderInner {
                         .message_sender
                         .send(WorkerMessage::Shutdown)
                         .await?;
-                    let new_count = worker_count.fetch_sub(1, std::sync::atomic::Ordering::Relaxed);
+                    let new_count = worker_count.fetch_sub(1, std::sync::atomic::Ordering::SeqCst);
```

Comment:
I think the culprit of adding 9 (11 total workers) is the relaxed memory ordering. I changed this to the sequential consistency model for easy reasoning. We can revisit if this becomes a performance bottleneck (the work_count is not update very frequently).